File: Complex.java
public class Complex {
    private double real;
    private double imaginary;
    // complex number formats:
    // A+Bi where B>1
    // A+i
    // A
    // Bi
    // i

    // constructors
    public Complex(double real, double imaginary){
        this.real=real;
        this.imaginary=imaginary;
    }

    // functions
    public Complex add(Complex o){
        double newReal=this.real+o.real;
        double newImaginary=this.imaginary+o.imaginary;
        return new Complex(newReal,newImaginary);
    }
    public Complex sub(Complex o){
        double newReal=this.real-o.real;
        double newImaginary= this.imaginary-o.imaginary;
        return new Complex(newReal,newImaginary);
    }
    public Complex mul(Complex o){
        double newReal=(this.real*o.real)+(this.imaginary*o.imaginary*-1);
        double newImaginary=(this.real*o.imaginary)+(this.imaginary*o.real);
        return new Complex(newReal,newImaginary);
    }
    public Complex div(Complex o){
        // o is the divisor
        // (this.real + this.imaginary) * (o.real + (o.imaginary * -1))
        double dividendReal= (this.real*o.real)+(this.imaginary*((o.imaginary*-1)*-1));
        double dividendImaginary=(this.real*(o.imaginary*-1))+(this.imaginary*(o.real));
        double divisorReal=(o.real*o.real)+(o.imaginary*(o.imaginary*-1)*-1);
        //double divisorImaginary=(o.real*(o.imaginary*-1))+(o.real*o.imaginary);
        double newReal= dividendReal/divisorReal;
        double newImaginary=dividendImaginary/divisorReal;
        return new Complex(newReal,newImaginary);
    }

    //int formal parameter
    public Complex add(double n){
        double newReal=this.real+n;
        return new Complex(newReal,this.imaginary);
    }
    public Complex sub(double n){
        double newReal=this.real-n;
        return new Complex(newReal,this.imaginary);
    }
    public Complex mul(double n){
        double newReal=this.real*n;
        double newImaginary=this.imaginary*n;
        return new Complex(newReal,newImaginary);
    }
    public Complex div(double n){
        double newReal=this.real/n;
        double newImaginary=this.imaginary/n;
        return new Complex(newReal,newImaginary);
    }
    // new toString method
    public String toString(){
        if(imaginary>0) {
            return "(" + real +"+"+imaginary+"i)";
        }
        else{
            return "(" + real +imaginary+"i)";
        }
    }
    public static void main(String[] args){
        Complex complex1=new Complex(1,2);
        Complex complex2=new Complex(1,3);
        System.out.println(complex1+" + "+complex2+" = "+complex1.add(complex2));
        System.out.println(complex1+" - "+complex2+" = "+complex1.sub(complex2));
        System.out.println(complex1+" * "+complex2+" = "+complex1.mul(complex2));
        System.out.println(complex1+" / "+complex2+" = "+complex1.div(complex2));
        //ints
        System.out.println(complex1+" + "+5+" = "+complex1.add(5));
        System.out.println(complex1+" - "+5+" = "+complex1.sub(5));
        System.out.println(complex1+" * "+5+" = "+complex1.mul(5));
        System.out.println(complex1+" / "+5+" = "+complex1.div(5));
    }
}

File: complex.py
class complex:
    def __init__(self, real, imaginary):
        self.real=real
        self.imaginary=imaginary
    def __add__(self, other):
        if isinstance(other, (int, float)):
            newReal=self.real+other;
            return complex(newReal,self.imaginary)
        elif isinstance(other, complex):
            newReal=self.real+other.real;
            newImaginary=self.imaginary+other.imaginary
            return complex(newReal,newImaginary)
        else:
            raise TypeError
    def __sub__(self, other):
        if isinstance(other, (int, float)):
            newReal=self.real-other;
            return complex(newReal,self.imaginary)
        elif isinstance(other, complex):
            newReal=self.real-other.real
            newImaginary=self.imaginary-other.imaginary
            return complex(newReal,newImaginary)
        else:
            raise TypeError
    def __mul__(self, other):
        if isinstance(other, (int, float)):
            newReal=self.real*other;
            newImaginary=self.imaginary*other
            return complex(newReal,newImaginary)
        elif isinstance(other, complex):
            newReal=(self.real*other.real)+(self.imaginary*other.imaginary*-1)
            newImaginary=(self.real*other.imaginary)+(other.real*self.imaginary)
            return complex(newReal,newImaginary)
        else:
            raise TypeError
    def __truediv__(self, other):
        if isinstance(other, (int, float)):
            newReal=self.real/other
            newImaginary=self.imaginary/other
            return complex(newReal,newImaginary)
        elif isinstance(other, complex):
            dividendReal= (self.real*other.real)+(self.imaginary*((other.imaginary*-1)*-1));
            dividendImaginary=(self.real*(other.imaginary*-1))+(self.imaginary*(other.real));
            divisorReal=(other.real*other.real)+(other.imaginary*(other.imaginary*-1)*-1);
            newReal= round(dividendReal/divisorReal,10);
            newImaginary=round(dividendImaginary/divisorReal,10);
            return complex(newReal,newImaginary)
        else:
            raise TypeError
    def __radd__(self, other):
        return self+other
    def __rsub__(self, other):
        return complex(other,0)-self
    def __rmul__(self, other):
        return self * other
    def __rtruediv__(self, other):
        return complex(other,0) / self
    def __str__(self):
        real_part=round(self.real,10)
        imaginary_part=round(self.imaginary,10)
        if(imaginary_part<0):
            return f'({real_part}{imaginary_part}i)'
        else:
            return f'({real_part}+{imaginary_part}i)'

if __name__ == '__main__':
    a=complex(1.0,2.0)
    b=complex(1.0,3.0)
    i = 5
    print('%s + %s = %s' % (a, b, a + b))
    print('%s - %s = %s' % (a, b, a - b))
    print('%s * %s = %s' % (a, b, a * b))
    print('%s / %s = %s' % (a, b, a / b))

    print('%s + %i = %s' % (a, i, a + i))
    print('%s - %i = %s' % (a, i, a - i))
    print('%s * %i = %s' % (a, i, a * i))
    print('%s / %i = %s' % (a, i, a / i))

    print('%i + %s = %s' % (i, a, i + a))
    print('%i - %s = %s' % (i, a, i - a))
    print('%i * %s = %s' % (i, a, i * a))
    print('%i / %s = %s' % (i, a, i / a))